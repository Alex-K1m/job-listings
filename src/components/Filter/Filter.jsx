import { arrayOf, func, string } from 'prop-types';
import React from 'react';

import Tag from '../Tag';
import styles from './Filter.module.scss';

const Filter = ({ tags, onRemoveTag, onClearTags }) => (
  <div className={styles.filter}>
    <ul className={styles.filterList}>
      {tags.map((tag) => (
        <li key={tag} className={styles.filterItem}>
          <Tag onDelete={onRemoveTag(tag)}>{tag}</Tag>
        </li>
      ))}
    </ul>
    <button type='button' className={styles.clearBtn} onClick={onClearTags}>
      Clear
    </button>
  </div>
);

Filter.propTypes = {
  tags: arrayOf(string).isRequired,
  onRemoveTag: func.isRequired,
  onClearTags: func.isRequired,
};

export default Filter;
