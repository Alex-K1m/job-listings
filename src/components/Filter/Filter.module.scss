@use '../colors' as c;
@use '../breakpoints' as brp;

.filter {
  align-items: center;
  background-color: #fff;
  border-radius: 0.5rem;
  box-shadow: 0 1.5rem 2rem -0.5rem c.$shadow;
  display: flex;
  justify-content: space-between;
  margin-top: -3.6rem;
  padding: 2rem 2.4rem 2rem 2rem;

  @include brp.sm-up {
    padding: 2rem 4rem;
  }

  &List {
    display: flex;
    flex-flow: wrap;
    margin: -0.8rem;
  }

  &Item {
    margin: 0.8rem;
  }
}

.clearBtn {
  background-color: transparent;
  border: 0;
  color: c.$text-secondary;
  cursor: pointer;
  font: 700 1.3rem/185% Spartan, sans-serif;
  letter-spacing: -0.0077em;
  margin-left: 1.2rem;
  padding: 0;

  &:hover {
    color: c.$primary;
    text-decoration: underline;
  }
}
