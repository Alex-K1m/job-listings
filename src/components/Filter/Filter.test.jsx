import React from 'react';
import renderer from 'react-test-renderer';

import Filter from './Filter';

describe('Filter', () => {
  it('renders correctly', () => {
    const tags = ['Junior', 'Frontend', 'React', 'JavaScript'];
    const json = renderer
      .create(
        <Filter tags={tags} onClearTags={jest.fn()} onRemoveTag={jest.fn()} />,
      )
      .toJSON();

    expect(json).toMatchSnapshot();
  });
});
