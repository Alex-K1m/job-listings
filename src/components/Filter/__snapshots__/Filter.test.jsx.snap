// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<div
  className="filter"
>
  <ul
    className="filterList"
  >
    <li
      className="filterItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          Junior
        </div>
      </div>
    </li>
    <li
      className="filterItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          Frontend
        </div>
      </div>
    </li>
    <li
      className="filterItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          React
        </div>
      </div>
    </li>
    <li
      className="filterItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          JavaScript
        </div>
      </div>
    </li>
  </ul>
  <button
    className="clearBtn"
    onClick={[MockFunction]}
    type="button"
  >
    Clear
  </button>
</div>
`;
